images/cfw/opt/x1plus/lib/python/x1plus/services/x1plusd/__init__.py

```diff
@@ -15,44 +20,192 @@
+        """
+
+        self.settings_keys = []
+
+        BASE_MODULE_DIR = os.path.dirname(os.path.abspath(__file__))
+        # Custom module folder could be in x1plus.services.x1plusd.custom, and symlinked to sd card?
+
+        directories = [BASE_MODULE_DIR]
+
+        for filename in os.listdir(BASE_MODULE_DIR):
+            full_path = os.path.join(BASE_MODULE_DIR, filename)
+            if os.path.isdir(full_path) and not filename.startswith("_"):
+                directories.append(full_path)
+
+        for directory in directories:
+            if not os.path.isdir(directory):
+                continue
+            for filename in os.listdir(directory):
+                try: 
+                    file_path = os.path.join(directory, filename)
+                    if not os.path.isfile(file_path):
+                        continue
+
+                    if filename.endswith(".py") and not filename.startswith("_"):
+                        module_data = module_docstring_parser(file_path, "x1plusd-module")
+                        if not module_data or not module_data.get("class_name", None):
+                            continue
+                        if not module_data.get("name", None):
+                            logger.warn(f"Missing required definitions in x1plusd module spec for {module_name}.")
+                            continue
+
+                        package = None
+                        if BASE_MODULE_DIR in directory:
+                            root_package = "x1plus.services.x1plusd"
+                            nested_dir = directory.replace(BASE_MODULE_DIR, "").replace("/", ".")
+                            if nested_dir: 
+                                # One level deep, i.e., 'custom'
+                                root_package = ".".join([root_package, nested_dir])
+                                root_package = root_package.replace("..", ".")
+                            package = root_package
+                        if not package:
+                            continue
+
+                        module, module_name = module_loader(file_path, package)
+
+                        """
+                        [x1plusd-module]
+                        name=name, required
+                        class_name=ClassName, required
+                        requires_router=bool, default: false
+                        start_func=function_name, default: task, set as 'null' if it does not have a start function
+                        [end]
+                        """
+
+                        if not module:
+                            continue
+                        if not hasattr(module, module_data.get("class_name")):
+                            logger.warn(f"Could not load {module_name} in x1plusd module loader. Class not found: {module_data.get("class_name")}")
+                            continue
+
+                        self.settings_keys.append(f"x1plusd.modules.{module_data.get("name")}")
```

Comment:
Purely used for listening to changes to restart all of x1plusd. It could have future use such as fetching all settings keys that x1plusd is currently watching for in regards to dynamic loading, whether enabled or disabled. But there are many other ways to do that.

Didn't like my name initially, will change